test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1355,9 +1370,8 @@ private void assertNoSnapshottedIndexCommit() throws Exception {
      * Asserts that the document history in Lucene index is consistent with Translog's on every index shard of the cluster.
      * This assertion might be expensive, thus we prefer not to execute on every test but only interesting tests.
      */
-    public void assertConsistentHistoryBetweenTranslogAndLuceneIndex() throws IOException {
-        final Collection<NodeAndClient> nodesAndClients = nodes.values();
-        for (NodeAndClient nodeAndClient : nodesAndClients) {
+    public synchronized void assertConsistentHistoryBetweenTranslogAndLuceneIndex() throws IOException {
```

Comment:
I think synchronized here is now unnecessary?

Reply:
Right dropping it :)